pkg/scheduler/internal/cache/node_tree.go

```diff
@@ -159,9 +162,7 @@ func (nt *NodeTree) resetExhausted() {
 
 // Next returns the name of the next node. NodeTree iterates over zones and in each zone iterates
 // over nodes in a round robin fashion.
-func (nt *NodeTree) Next() string {
-	nt.mu.Lock()
```

Comment:
Can you confirm that the lock here should be deleted?

Reply:
Yes, that is actually the main goal of this PR, removing the contention here.